lib/bundles/inspec-compliance/api.rb

```diff
@@ -251,11 +251,55 @@ def self.server_version_from_config(config)
     end
 
     def self.determine_server_type(url, insecure)
-      if Compliance::HTTP.get(url + '/compliance/version', nil, insecure).code == '401'
-        :automate
-      elsif Compliance::HTTP.get(url + '/api/version', nil, insecure).code == '200'
-        :compliance
+      return :automate if target_is_automate_server?(url, insecure)
+      return :compliance if target_is_compliance_server?(url, insecure)
+      Inspec::Log.debug('Could not determine server type using known endpoints')
+      nil
+    end
+
+    def self.target_is_automate_server?(url, insecure)
+      automate_endpoint = '/compliance/version'
+      response = Compliance::HTTP.get(url + automate_endpoint, nil, insecure)
+      case response.code
+      when '401'
+        Inspec::Log.debug(
+          "Received 401 from #{url}#{automate_endpoint} " \
+          'assuming target is a Chef Automate instance',
+        )
+        return true
+      when '200'
+        # Chef Automate currently returns 401 for `/compliance/version` but some
+        # versions of OpsWorks Chef Automate return 200 and a Chef Manage page
+        # when unauthenticated requests are received.
+        if response.body.include?('Are You Looking For the Chef Server?')
+          Inspec::Log.debug(
+            "Received 200 from #{url}#{automate_endpoint} " \
+            'assuming target is an OpsWorks Chef Automate instance',
+          )
+          return true
+        else
+          Inspec::Log.debug(
+            "Received 200 from #{url}#{automate_endpoint} " \
+            'but did not receive the Chef Manage page. ' \
+            'Continuing with detection attempts',
+          )
+          return false
+        end
       end
```

Comment:
We'll return `nil` if we get a non-200 / non-401 status code. I think we should have a default `else` in the `case` statement that's just `false`

And if you do that, you can eliminate the `return` statements in 269, 279, and 286 and just have them be `true` or `false` accordingly. The `return`s are superfluous.

Reply:
Ah! Great catch. I will make it so.